mws/resource_mws_workspaces.go

```diff
@@ -310,7 +321,7 @@ func (a WorkspacesAPI) UpdateRunning(ws Workspace, timeout time.Duration) error
 	if err != nil {
 		return err
 	}
-	return a.WaitForRunning(ws, timeout)
+	return a.WaitForExpectedStatus(ws, WorkspaceStatusRunning, timeout)
```

Comment:
Let's set the expected workspace status that is provided by the user in this request.